pkg/tenant/tenants.go

```diff
@@ -0,0 +1,76 @@
+package tenant
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/tenant/api/v1"
+	"github.com/rotationalio/ensign/pkg/tenant/db"
+	"github.com/rs/zerolog/log"
+)
+
+// TenantCreates creates adds a new tenant to the database
+func (s *Server) TenantCreate(c *gin.Context) {
+	var (
+		err error
+		t   *api.Tenant
+	)
+
+	// TODO: Add authentication and authorization middleware
+
+	if err = c.BindJSON(&t); err != nil {
+		log.Warn().Err(err).Msg("could not bind tenant create request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not bind request"))
+	}
+
+	if t.ID != "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrTenantIDRequired))
+	}
+
+	if t.Name == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("tenant name is required"))
+	}
+
+	if t.EnvironmentType == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("environment type is required"))
+	}
+
+	tenantID := ulid.Make()
+	created := time.Now()
+	modified := created
```

Comment:
Is it necessary to do this work here? Won't the `db` package do this?

Reply:
I got this mixed up with a previous comment about updating the response with what's generated from the db and will make the change.